client/lib/signup/flow-controller.ts

```diff
@@ -349,7 +349,7 @@ export default class SignupFlowController {
 
 	_getStoredDependencies() {
 		const requiredDependencies = flatMap( this._flow.steps, stepName =>
-			get( steps, [ stepName, 'providesDependencies' ], [] )
+			get( steps[ stepName ], 'providesDependencies', [] )
```

Comment:
I know it's a bit more verbose, but how about we deal with the problem by removing `_.get` from this line altogether, and instead use:

```js
( steps && steps[ stepName ] && steps[ stepName ].providesDependencies ) || []
```

As long as `providesDependencies` can't legitimately hold any non-nullish falsey value (that is, `''`, `NaN`, `false`, or `0`), then this should work correctly.

Once we have TypeScript 3.7, we can refactor this to the much shorter (and stricter):

```js
steps?.[ stepName ]?.providesDependencies ?? []
```

Reply:
Also, I see line 229 using the exact same `_.get` call. Wouldn't this be an issue there too?

Reply:
Happy to remove uses of `_.get` :)

The original error wasn't actually with `requiredDependencies`, it was on the next statement. TS bent over backwards and infered some zany type for it (not a `string[]`) which is why it couldn't be used with `_.pick`.

It had no issue with what was on line 229  because the code that followed was happy to use the zany type.

In either case I've updated both cases to the code you suggested, `string[]` is what I want it to be :)

Reply:
That makes perfect sense, thank you for the clarification! 🙂

Reply:
It looks like you may have forgotten to push your changes, but I didn't want to delay you any further, so I approved it with the above discussion in mind.

Reply:
Whoops, thanks :)